lte/gateway/c/core/oai/tasks/mme_app/mme_app_state_manager.cpp

```diff
@@ -140,33 +144,16 @@ void MmeNasStateManager::create_hashtables() {
   state_cache_p->mme_ue_contexts.s11_teid2mme_ueid_map.set_name(
       MME_S11_TEID2MME_UE_ID_MAP_NAME);
 
-  bstring b = bfromcstr(UE_ID_UE_CTXT_TABLE_NAME);
-  state_ue_ht = hashtable_ts_create(max_ue_htbl_lists_, nullptr,
-                                    mme_app_state_free_ue_context, b);
-
-  if (!(state_ue_ht->lock_attr = (pthread_mutexattr_t*)calloc(
-            max_ue_htbl_lists_, sizeof(pthread_mutexattr_t)))) {
-    free_wrapper((void**)&state_ue_ht->lock_nodes);
-    free_wrapper((void**)&state_ue_ht->nodes);
-    free_wrapper((void**)&state_ue_ht->name);
-    free_wrapper((void**)&state_ue_ht);
-    return;
-  }
-
-  for (int i = 0; i < max_ue_htbl_lists_; i++) {
-    pthread_mutexattr_init(&state_ue_ht->lock_attr[i]);
-    pthread_mutexattr_settype(&state_ue_ht->lock_attr[i],
-                              PTHREAD_MUTEX_RECURSIVE);
-    pthread_mutex_init(&state_ue_ht->lock_nodes[i], &state_ue_ht->lock_attr[i]);
-  }
+  mme_app_state_ue_map.map =
+      new google::protobuf::Map<uint32_t, struct ue_mm_context_s*>();
+  mme_app_state_ue_map.set_name(MME_UE_ID2UE_CTXT_MAP_NAME);
```

Comment:
This is not needed as UE context will be released as part of mme_remove_ue_context(). Adding mme_app_state_free_ue_context() as callback here leads to double free error

Reply:
1. Why double free error not observed earlier?
2. Any memory leaks with mme restart or with all services stop?

Reply:
1. This was not observed earlier because mme_app_state_free_ue_context function is not getting called at all even though it is passed as a callback function in hashtable_ts_create. Hence it is getting freed only once as part of mme_remove_ue_context(). 
2. There are no memory leaks

Reply:
Remove fun, mme_app_state_free_ue_context() is not used

Reply:
Added back this function and passed false to remove() so that ue_context does not get freed here